test/e2e/cli/pkg_unsource_test.go

```diff
@@ -0,0 +1,362 @@
+
+					// Read file content
+					readBytes, err := os.ReadFile(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+
+					// Marshal the yaml file into a map
+					var cfgMap map[string]interface{}
+					err = yaml.Unmarshal([]byte(readBytes), &cfgMap)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file contains the default manifests
+					cfgMapUnikernel, ok := cfgMap["unikraft"].(map[string]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Cast the manifests list to an array
+					cfgMapUnikernelManifests, ok := cfgMapUnikernel["manifests"].([]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Check if the default manifests are removed
+					Expect(cfgMapUnikernelManifests).To(HaveLen(0))
+
+					// Check if stdout is empty
+					Expect(stdout.String()).To(BeEmpty())
+				})
+			})
+
+			When("the config does not contain the link or it doesn't exist", func() {
+				BeforeEach(func() {
+					// Save the config file by moving it to a temporary location
+					err := os.Rename(cfg.Path(), cfg.Path()+".tmp")
+					Expect(err).ToNot(HaveOccurred())
+				})
+				AfterEach(func() {
+					// Restore the config file
+					err := os.Rename(cfg.Path()+".tmp", cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+				})
+				It("should do nothing and print a warning", func() {
+					cmd.Args = append(cmd.Args, "https://example.com")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Read the config file
+					osFile, err := os.Open(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+					defer osFile.Close()
+
+					osFileInfo, err := osFile.Stat()
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file is not empty
+					Expect(osFileInfo.Size()).To(BeNumerically(">", 0))
+
+					// Check if the config file has 600 permissions
+					Expect(osFileInfo.Mode().Perm()).To(Equal(os.FileMode(0o600)))
+
+					// Read file content
+					readBytes, err := os.ReadFile(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+
+					// Marshal the yaml file into a map
+					var cfgMap map[string]interface{}
+					err = yaml.Unmarshal([]byte(readBytes), &cfgMap)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file contains the default manifests
+					cfgMapUnikernel, ok := cfgMap["unikraft"].(map[string]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Cast the manifests list to an array
+					cfgMapUnikernelManifests, ok := cfgMapUnikernel["manifests"].([]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Check if the default manifests are still there
+					Expect(cfgMapUnikernelManifests).To(HaveLen(1))
+					Expect(cfgMapUnikernelManifests[0]).To(Equal("https://manifests.kraftkit.sh/index.yaml"))
+
+					// Check if stdout contains the warning
+					Expect(stdout.String()).To(MatchRegexp(`^{"level":"warning","msg":"manifest not found: https://example\.com"}\n$`))
```

Comment:
nit: We could as well use `Equal()` here if we are gating the string between `^$`.